server/partition.go

```diff
@@ -802,6 +817,31 @@ func (p *partition) messageProcessingLoop(recvChan <-chan *nats.Msg, stop <-chan
 		p.mu.Unlock()
 
 		m := natsToProtoMessage(msg, leaderEpoch)
+
+		if p.encryptionHandler != nil {
+			// Encrypt value
```

Comment:
Related to above comment, I wonder if this is the appropriate place to do encryption or if it should live somewhere else, such as a log wrapper.

Reply:
Agree. my comments are written above